release-22.2:     changefeedccl: Fix setting propagation flake

miretskiy opened the pull request:
Backport 1/2 commits from #110601.

/cc @cockroachdb/release

---

Setting changes are asynchronous, particularly in tenants.
Set the feature setting directly instead.

Fixes #110300

Release note: None
Release justification: Test only fix

Commit:
changefeedccl: Fix setting propagation flake

Setting changes are asynchronous, particularly in tenants.
Set the feature setting directly instead.

Fixes #110300

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/110636)
<!-- Reviewable:end -->

jayshrivastava commented:
@miretskiy Do we have an issue open to audit and fix all the times we do this? I think override settings like this very often.

Commit:
changefeedccl: Timeout test if error does not happen soon

Add a timeout when expecting an error to occur in tests.

Informs #110300

Release note: None

Commit:
changefeedccl: Add a test utility to timeout when waiting for error

Add `ExpectErrWithTimeout` test utility which times out
if the error does not occur withing expected threshold.
This utility function differes from ExpectErrSucceedsSoon
in that it does not retry the call -- it merely expects
an error to be returned withing timeout threshold.

Informs #110300

Release note: None

miretskiy commented:
> @miretskiy Do we have an issue open to audit and fix all the times we do this? I think override settings like this very often.

I don't have an issue to fix this on our side; I think we'll continue working on this on one-off basis.
https://github.com/cockroachdb/cockroach/issues/110560 is the issue to make sure settings are propagated in multi tenant.

jayshrivastava reviewed (approved):
Btw pkg/testutils/fuzz/BUILD.bazel is empty
